src/ws-dropdown.js

```diff
@@ -0,0 +1,252 @@
+    };
+    // Set states to items in item list for passed values
+    this.state.items.forEach(item => {
+      if (this.state.value.find(val => val.label == item.label)) {
+        item.selected = true;
+        item.stored = true;
+      }
+    });
+  }
+
+  /**
+   * Called by React to get the types of the child context values
+   * @returns {Object}
+   */
+  getChildContext() {
+    return {
+      multiple: this.props.multiple
+    }
+  }
+
+  /**
+   * Start listening for clicks in window
+   * @void
+   */
+  componentDidMount() {
+    window.addEventListener('click', this.onDocumentClick.bind(this));
+  }
+
+  /**
+   * Stop listening for clicks in window
+   * @void
+   */
+  componentWillUnmount() {
+    window.removeEventListener('click', this.onDocumentClick.bind(this));
+  }
+
+  /**
+   * Used to convert the items if they are strings into the required object structure
+   * @param items {Array<String|Object>}
+   * @returns {Array<Object>}
+   */
+  enrichItems(items) {
+    return items.map(item => {
+      let enriched = typeof item === 'object' ? item : {label: item};
+      if (enriched.children) {
+        enriched.children = this.enrichItems(enriched.children);
+      }
+      return enriched;
+    });
+  }
+
+  /**
+   * Handles data propagation from child elements
+   * @param type {String} Either change for value changes or change-size which will be emitted on menu changes
+   * @param data {Object|Number} Either new value or height of new menu
+   * @void
+   */
+  handlePropagation(type, data) {
+    if (type === 'change') {
+      this.close();
+      this.setValue(data);
+      this.element.dispatchEvent(new CustomEvent('change', { detail: data }));
+    }
+    else if (type === 'change-size') {
+      this.adjustSize(data);
+    }
+  }
+
+  /**
+   * Set the value of the dropdown and update the display text if the trigger element is a select
+   * @param value {Object|Array<Object>}
+   * @void
+   */
+  setValue(value) {
+    this.state.value = value;
+    // Check if we have to update the text value
+    if (this.props.type === 'select') {
+      if (Array.isArray(this.state.value)) {
+        this.state.text = this.state.value.map(item => item.label).join(', ');
+      }
+      else {
+        this.state.text = this.state.value.label;
+      }
+      this.setState(this.state);
+    }
+  }
+
+  /**
+   * Handles click on document body to close the dropdown if clicked elsewhere
+   * @param event {MouseEvent}
+   * @void
+   */
+  onDocumentClick(event) {
+    let element = event.target;
+    while (element && this.element !== element) {
+      element = element.parentNode;
+    }
+    // Element will be empty if not clicked into the current filter dialog
+    if (!element) {
+      this.close();
+    }
+  }
+
+  /**
+   * Opens the dropdown
+   * @void
+   */
+  open() {
+    if (this.state.opened) {
+      return;
+    }
+    this.state.opened = true;
+    this.dropdownContainer.style.height = 0;
```

Comment:
I have `ref={element => { this.dropdownContainer = element; }}` which should do the same. But nice to know that the other way works as well.
Can you tell me how this could happen or how to reproduce?